woocommerce/woocommerce-ios
# [Woo POS] UI: Product cards. Add and remove products from `CartView`

Partially addresses https://github.com/woocommerce/woocommerce-ios/issues/12703
Closes #12722
Closes #12721

## Description

This PR updates the `Product Card View` closer to the proposed initial designs and adds some some basic communication logic between the `ProductListView` and `CartView`, since we need a minimum behaviour to add products from one to the other, as well as to remove items from the cart. All changes are managed by the `PointOfSaleDashboardViewModel` which acts as a the single source of truth.

![Simulator Screen Recording - iPad Pro (12 9-inch) (6th generation) - 2024-05-14 at 18 31 45](https://github.com/woocommerce/woocommerce-ios/assets/3812076/dc1d77e6-c391-4aae-bdf1-1a8cbe707357)

I don't think we want to polish much the designs right now, but I'm not sure neither how much is "good enough". I've stopped when the core components were working and we had a minimum of design, but happy to iterate as needed 🙇 

**Question:** Do we want to start to add some basic unit tests? Same reason as above: I don't want to go too deep since we do not have data layer, but I see the benefit of test-driving the expected behaviour a bit at this level.

## Changes:
- Created views: `ProductCardView`, `QuantityBadgeView` (non functional), `PlusButtonView`
- Created model: `CartProduct`. This is a simple struct that represents each product added to the cart. Is just a wrapper for `Product` with an unique ID so we can identify it across similar products.
- Updated the `productsInCart` observable property type, which has changed to be of type `CartProduct`, so we can keep track of the `Product` list and the `CartProducts` on cart independently, but sharing the same source of truth.

## Testing instructions
* In the POS view:
  * Tap on the `+` button on the product cards, observe that the correct product is added to the Cart.
  * Tap on the `x` button on the cart, observe that the correct product is removed from the Cart.
